blocks/api/raw-handling/test/shortcode-converter.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { equal } from 'assert';
+
+/**
+ * Internal dependencies
+ */
+import segmentHTMLToShortcodeBlock from '../shortcode-converter';
+import {
+	getBlockTransforms,
+} from '../../factory';
+import {
+	getPhrasingContentSchema,
+} from '../utils';
+
+describe( 'segmentHTMLToShortcodeBlock', () => {
+	const blockTypes = [
+		{
+			name: 'core/paragraph',
+			transforms: {
+				from: [
+					{
+						type: 'raw',
+						// Paragraph is a fallback and should be matched last.
+						priority: 20,
+						selector: 'p',
+						schema: {
+							p: {
+								children: getPhrasingContentSchema(),
+							},
+						},
+					},
+				],
+			},
+		},
+		{
+			name: 'core/shortcode',
+			transforms: {
+				from: [
+					{
+						type: 'shortcode',
+						// Per "Shortcode names should be all lowercase and use all
+						// letters, but numbers and underscores should work fine too.
+						// Be wary of using hyphens (dashes), you'll be better off not
+						// using them." in https://codex.wordpress.org/Shortcode_API
+						// Require that the first character be a letter. This notably
+						// prevents footnote markings ([1]) from being caught as
+						// shortcodes.
+						tag: '[a-z][a-z0-9_-]*',
+						attributes: {
+							text: {
+								type: 'string',
+								shortcode: ( attrs, { content } ) => {
+									return content;
+								},
+							},
+						},
+						priority: 20,
+					},
+				],
+			},
+		},
+	];
+	const transformsFrom = getBlockTransforms( 'from', undefined, blockTypes );
+
+	it( 'should convert a standalone shortcode between two paragraphs', () => {
+		const original = `<p>Foo</p>
+
+[foo bar="apple"]
+
+<p>Bar</p>`;
+		const expected = [
+			`<p>Foo</p>
+
+`,
+			{
+				attributes: {},
+				innerBlocks: [],
+				isValid: true,
+				name: 'core/shortcode',
+				uuid: 'invalid',
```

Comment:
`uuid` being a random value is the cause of the test failure right now. What's the best way of addressing?

Reply:
Not entirely sure it works in this specific usage, but perhaps `expect.any( String )` ?

http://jestjs.io/docs/en/expect#expectanyconstructor

Reply:
> Not entirely sure it works in this specific usage

Yeah, I suppose I could make the assertions a bit more verbose and inspect each part of the array.

Another idea is to create a wrapper for the `uuid` function that lets us inject some state for the purposes of testing.

Reply:
`expect.any( String )` is better, we don't care about what `uuid` produces in this context.